extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/job/Job.java

```diff
@@ -0,0 +1,218 @@
+        scheduleEvent(thingUID, astroHandler, range.getEnd(), EVENT_END, channelId);
+    }
+
+    /**
+     * Schedules {@link Planet} events
+     *
+     * @param thingUID
+     *            the {@link Thing} UID
+     * @param astroHandler
+     *            the {@link AstroThingHandler} instance
+     * @param eventAt
+     *            the {@link Calendar} instance denoting scheduled instant
+     */
+    public static void schedulePublishPlanet(String thingUID, AstroThingHandler astroHandler,
+            Calendar eventAt) {
+        boolean thingNull = checkNull(thingUID, "Thing UID is null");
+        boolean astroHandlerNull = checkNull(astroHandler, "AstroThingHandler is null");
+        boolean eventAtNull = checkNull(eventAt, "Scheduled Instant is null");
+
+        if (thingNull || astroHandlerNull || eventAtNull) {
+            return;
+        }
+        Job publishJob = new PublishPlanetJob(thingUID);
+        schedule(thingUID, astroHandler, publishJob, eventAt);
+    }
+
+    /**
+     * Schedules {@link SunPhaseJob}
+     *
+     * @param thingUID
+     *            the {@link Thing} UID
+     * @param astroHandler
+     *            the {@link AstroThingHandler} instance
+     * @param sunPhaseName
+     *            {@link SunPhaseName} instance
+     * @param eventAt
+     *            the {@link Calendar} instance denoting scheduled instant
+     */
+    public static void scheduleSunPhase(String thingUID, AstroThingHandler astroHandler,
+            SunPhaseName sunPhaseName, Calendar eventAt) {
+        boolean thingNull = checkNull(thingUID, "Thing UID is null");
+        boolean astroHandlerNull = checkNull(astroHandler, "AstroThingHandler is null");
+        boolean sunPhaseNull = checkNull(sunPhaseName, "Sun Phase Name is null");
+        boolean eventAtNull = checkNull(eventAt, "Scheduled Instant is null");
+
+        if (thingNull || astroHandlerNull || sunPhaseNull || eventAtNull) {
+            return;
+        }
+        Job sunPhaseJob = new SunPhaseJob(thingUID, sunPhaseName);
+        schedule(thingUID, astroHandler, sunPhaseJob, eventAt);
+    }
+
+    /**
+     * Checks that the specified object reference is not {@code null} and logs a
+     * customized message if it is. This method is designed primarily for doing
+     * parameter validation in methods and constructors with multiple
+     * parameters, as demonstrated below: <blockquote>
+     *
+     * <pre>
+     * public Foo(Bar bar, Baz baz) {
+     *     checkNull(bar, "bar is null");
+     *     checkNull(baz, "baz is null");
+     * }
+     * </pre>
+     *
+     * </blockquote>
+     *
+     * @param obj
+     *            the object reference to check for {@code null}
+     * @param message
+     *            detail message to be used in the event
+     * @return {@code true} if {@code null} otherwise {@code false}
+     */
+    public static <T> boolean checkNull(T obj, String message) {
```

Comment:
In the whole framework we do not log all cases where an object can be null. Please do so if this results in a serious (permanent) error, otherwise we do NOT need a logging of null objects.